src/sentry/auth/access.py

```diff
@@ -319,21 +336,184 @@ def has_any_project_scope(self, project: Project, scopes: Collection[str]) -> bo
 
         return False
 
-    def to_django_context(self) -> Mapping[str, bool]:
-        return {s.replace(":", "_"): self.has_scope(s) for s in settings.SENTRY_SCOPES}
 
+@dataclass
+class ApiAccess(Access):
+    api_user_organization_context: ApiUserOrganizationContext
+    requested_scopes: Iterable[str] | None
+    auth_state: ApiAuthState
```

Comment:
AuthState captures both Sso state as well as permissions for super users, which are also control silo only.